policies/common/Silence.yaml

```diff
@@ -0,0 +1,19 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: inject-heartbeat-negative-matcher-to-let-them-through
+spec:
+  rules:
+    - name: inject-heartbeat-negative-matcher-to-let-them-through
+      match:
+        resources:
+          kinds:
+            - monitoring.giantswarm.io/v1alpha1/Silence
+      mutate:
+        patchStrategicMerge:
+          spec:
+            matchers:
```

Comment:
What exactly is the expected outcome here? Like do you just want to make sure this additional matcher exists?

`inject-heartbeat-negative-matcher-to-let-them-through` doesn't really tell me much :sweat_smile:

Reply:
Yes, I want the webhook to add the negative matcher. I'm not really sure how to name it :D Basically, this should enforce a matcher that lets heartbeats through if we silence a cluster

Reply:
WDYT of `do-not-match-heartbeat-alert-when-silencing-a-cluster` ?

Reply:
because naming is hard here are some ideas:

- `always-allow-heartbeat`
- `heartbeat-on`
- `heartbeat-matcher`
- `skip-heartbeat`

Reply:
I do prefer always-allow-heartbeat

Reply:
Okay - so every silence should include a matcher for the Heartbeat? If I understand that correctly?

Have you tested this already? Maybe adding some integration test could be nice (I can help :D )

Reply:
I'm waiting for my PR in API extensions to be merged first before I can test it but I can give it a go to pytest :D I will try to add one this mornign and I will ping you for help if needed :D